File: runtime/CSharp/runtime/CSharp/Antlr4.Runtime/Atn/LexerChannelAction.cs
/*
 * [The "BSD license"]
 *  Copyright (c) 2013 Terence Parr
 *  Copyright (c) 2013 Sam Harwell
 *  All rights reserved.
 *
 *  Redistribution and use in source and binary forms, with or without
 *  modification, are permitted provided that the following conditions
 *  are met:
 *
 *  1. Redistributions of source code must retain the above copyright
 *     notice, this list of conditions and the following disclaimer.
 *  2. Redistributions in binary form must reproduce the above copyright
 *     notice, this list of conditions and the following disclaimer in the
 *     documentation and/or other materials provided with the distribution.
 *  3. The name of the author may not be used to endorse or promote products
 *     derived from this software without specific prior written permission.
 *
 *  THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
 *  IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
 *  OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
 *  IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT,
 *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
 *  NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
 *  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
 *  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
 *  THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
using Antlr4.Runtime;
using Antlr4.Runtime.Atn;
using Antlr4.Runtime.Misc;
using Antlr4.Runtime.Sharpen;

namespace Antlr4.Runtime.Atn
{
    /// <summary>
    /// Implements the
    /// <c>channel</c>
    /// lexer action by calling
    /// <see cref="Lexer.Channel"/>
    /// with the assigned channel.
    /// </summary>
    /// <author>Sam Harwell</author>
    /// <since>4.2</since>
    public sealed class LexerChannelAction : ILexerAction
    {
        private readonly int channel;

        /// <summary>
        /// Constructs a new
        /// <paramref name="channel"/>
        /// action with the specified channel value.
        /// </summary>
        /// <param name="channel">
        /// The channel value to pass to
        /// <see cref="Lexer.Channel"/>
        /// .
        /// </param>
        public LexerChannelAction(int channel)
        {
            this.channel = channel;
        }

        /// <summary>
        /// Gets the channel to use for the
        /// <see cref="Antlr4.Runtime.IToken"/>
        /// created by the lexer.
        /// </summary>
        /// <returns>
        /// The channel to use for the
        /// <see cref="Antlr4.Runtime.IToken"/>
        /// created by the lexer.
        /// </returns>
        public int Channel
        {
            get
            {
                return channel;
            }
        }

        /// <summary><inheritDoc/></summary>
        /// <returns>
        /// This method returns
        /// <see cref="LexerActionType.Channel"/>
        /// .
        /// </returns>
        public LexerActionType ActionType
        {
            get
            {
                return LexerActionType.Channel;
            }
        }

        /// <summary><inheritDoc/></summary>
        /// <returns>
        /// This method returns
        /// <see langword="false"/>
        /// .
        /// </returns>
        public bool IsPositionDependent
        {
            get
            {
                return false;
            }
        }

        /// <summary>
        /// <inheritDoc/>
        /// <p>This action is implemented by calling
        /// <see cref="Lexer.Channel"/>
        /// with the
        /// value provided by
        /// <see cref="Channel()"/>
        /// .</p>
        /// </summary>
        public void Execute(Lexer lexer)
        {
            lexer.Channel = channel;
        }

        public override int GetHashCode()
        {
            int hash = MurmurHash.Initialize();
            hash = MurmurHash.Update(hash, (int)(ActionType));
            hash = MurmurHash.Update(hash, channel);
            return MurmurHash.Finish(hash, 2);
        }

        public override bool Equals(object obj)
        {
            if (obj == this)
            {
                return true;
            }
            else
            {
                if (!(obj is Antlr4.Runtime.Atn.LexerChannelAction))
                {
                    return false;
                }
            }
            return channel == ((Antlr4.Runtime.Atn.LexerChannelAction)obj).channel;
        }

        public override string ToString()
        {
            return string.Format("channel({0})", channel);
        }
    }
}

File: runtime/Swift/Sources/Antlr4/atn/LexerChannelAction.swift
/// 
/// Copyright (c) 2012-2017 The ANTLR Project. All rights reserved.
/// Use of this file is governed by the BSD 3-clause license that
/// can be found in the LICENSE.txt file in the project root.
/// 


/// 
/// Implements the `channel` lexer action by calling
/// _org.antlr.v4.runtime.Lexer#setChannel_ with the assigned channel.
/// 
/// -  Sam Harwell
/// -  4.2
/// 

public final class LexerChannelAction: LexerAction, CustomStringConvertible {
    fileprivate let channel: Int

    /// 
    /// Constructs a new `channel` action with the specified channel value.
    /// - parameter channel: The channel value to pass to _org.antlr.v4.runtime.Lexer#setChannel_.
    /// 
    public init(_ channel: Int) {
        self.channel = channel
    }

    /// 
    /// Gets the channel to use for the _org.antlr.v4.runtime.Token_ created by the lexer.
    /// 
    /// - returns: The channel to use for the _org.antlr.v4.runtime.Token_ created by the lexer.
    /// 
    public func getChannel() -> Int {
        return channel
    }

    /// 
    /// 
    /// - returns: This method returns _org.antlr.v4.runtime.atn.LexerActionType#CHANNEL_.
    /// 

    public override func getActionType() -> LexerActionType {
        return LexerActionType.channel
    }

    /// 
    /// 
    /// - returns: This method returns `false`.
    /// 

    public override func isPositionDependent() -> Bool {
        return false
    }

    /// 
    /// 
    /// 
    /// This action is implemented by calling _org.antlr.v4.runtime.Lexer#setChannel_ with the
    /// value provided by _#getChannel_.
    /// 

    public override func execute(_ lexer: Lexer) {
        lexer.setChannel(channel)
    }


    public override func hash(into hasher: inout Hasher) {
        hasher.combine(getActionType())
        hasher.combine(channel)
    }

    public var description: String {
        return "channel\(channel)"
    }

}


public func ==(lhs: LexerChannelAction, rhs: LexerChannelAction) -> Bool {

    if lhs === rhs {
        return true
    }


    return lhs.channel == rhs.channel
}
